packages/block-editor/src/components/responsive-block-control/index.js

```diff
@@ -0,0 +1,85 @@
+/**
+ * External dependencies
+ */
+import { uniqueId } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+
+import { Fragment, useState } from '@wordpress/element';
+
+import {
+	ToggleControl,
+} from '@wordpress/components';
+
+export function ResponsiveBlockControlLabel( { property, device } ) {
+	const accessibleLabel = sprintf( __( 'Controls the %s property for %s devices.' ), property, device );
+	const descId = uniqueId();
+	return (
+		<Fragment>
+			<span aria-describedby={ `rbc-desc-${ descId }` }>
+				{ device }
+			</span>
+			<span className="screen-reader-text" id={ `rbc-desc-${ descId }` }>{ accessibleLabel }</span>
+		</Fragment>
+	);
+}
+
+function ResponsiveBlockControl( props ) {
+	const { legend, property, toggleLabel, responsiveControlsActive = false, renderDefaultControl, defaultLabel = __( 'All' ), devices = [ __( 'Desktop' ), __( 'Tablet' ), __( 'Mobile' ) ], renderResponsiveControls } = props;
+
+	const [ isResponsiveMode, setIsResponsiveMode ] = useState( responsiveControlsActive );
+
+	if ( ! legend || ! property || ! renderDefaultControl ) {
+		return null;
+	}
+
+	const toggleControlLabel = toggleLabel || sprintf( __( 'Use the same %s on all screensizes.' ), property );
+
+	const toggleHelpText = __( 'Toggle between using the same value for all screen sizes or using a unique value per screen size.' );
+
+	const defaultControl = renderDefaultControl( defaultLabel );
+
+	const defaultResponsiveControls = () => {
+		return devices.map( ( deviceLabel, index ) => (
+			<Fragment key={ index }>
+				{ renderDefaultControl( deviceLabel ) }
+			</Fragment>
+		)
+		);
+	};
+
+	const handleToggle = ( isChecked ) => {
+		setIsResponsiveMode( ! isChecked );
+	};
+
+	return (
+
+		<fieldset className="block-editor-responsive-block-control">
+			<legend className="block-editor-responsive-block-control__legend">{ legend }</legend>
+
+			<div className="block-editor-responsive-block-control__inner">
+				<ToggleControl
+					className="block-editor-responsive-block-control__toggle"
+					label={ toggleControlLabel }
+					checked={ ! isResponsiveMode }
+					onChange={ handleToggle }
+					help={ toggleHelpText }
+				/>
+
```

Comment:
Why hiding instead of rendering one option or the other depending on isResponsiveMode?

Reply:
I was thinking about a11y. I was thinking if you were focused on one set of controls and then you toggled to the other "mode" (eg: "Default" -> "Responsive"). I felt it might be a bit confusing or jarring to have the original controls completely removed from the DOM and a load of new controls suddenly appear.

What do you think @afercia?

Reply:
>  I felt it might be a bit confusing or jarring to have the original controls completely removed from the DOM and a load of new controls suddenly appear.

If we leave things on the dom, screen reader users may still be able to interact with the controls? If that's the case, they will interact with controls that make nothing that seems worst.
We have at least another case exactly like this the hasParallax prop in the Cover block. That case does not renders on the dom so I guess we should follow the same approach here. If we find something better from the a11y point of view in this PR I think as a follow-up we should update the other PR.

Reply:
Ok noted. Will change.

Reply:
Fixed.